### ISSUE TITLE
Deprecation warning for lr scheduler verbose init arg

### ISSUE THREAD

**Issue by scarere** [MIC-DKFZ/nnUNet#2680]
The verbose argument of PolyLRScheduler's parent class is now deprecated. The argument need to not be passed to prevent the deprecation warning.

**Comment by scarere** [comment]
Its an easy fix.

```python
from torch.optim.lr_scheduler import _LRScheduler


class PolyLRScheduler(_LRScheduler):
    def __init__(self, optimizer, initial_lr: float, max_steps: int, exponent: float = 0.9, current_step: int = None):
        self.optimizer = optimizer
        self.initial_lr = initial_lr
        self.max_steps = max_steps
        self.exponent = exponent
        self.ctr = 0
        # Below is the only line that was changed. Removed the third argument which was specifying verbose as False
        super().__init__(optimizer, current_step if current_step is not None else -1)

    def step(self, current_step=None):
        if current_step is None or current_step == -1:
            current_step = self.ctr
            self.ctr += 1

        new_lr = self.initial_lr * (1 - current_step / self.max_steps) ** self.exponent
        for param_group in self.optimizer.param_groups:
            param_group['lr'] = new_lr
```

**Comment by scarere** [comment]
Furthermore while you're at it, the _LRScheduler class only exists for backwards compatibility. It is literally defined in the pytorch source as:
```python
class _LRSchedulre(LRScheduler):
    pass
```

You might as well update the class parent as _LRScheduler is likely to become deprecated in the future.

```python
from torch.optim.lr_scheduler import LRScheduler


class PolyLRScheduler(LRScheduler):
    def __init__(self, optimizer, initial_lr: float, max_steps: int, exponent: float = 0.9, current_step: int = None):
        self.optimizer = optimizer
        self.initial_lr = initial_lr
        self.max_steps = max_steps
        self.exponent = exponent
        self.ctr = 0
        # Below is the only line that was changed. Removed the third argument which was specifying verbose as False
        super().__init__(optimizer, current_step if current_step is not None else -1)

    def step(self, current_step=None):
        if current_step is None or current_step == -1:
            current_step = self.ctr
            self.ctr += 1

        new_lr = self.initial_lr * (1 - current_step / self.max_steps) ** self.exponent
        for param_group in self.optimizer.param_groups:
            param_group['lr'] = new_lr
```

**Comment by mrokuss** [comment]
Hey thanks!

Do you want to file a PR so we can merge it?

Best,
Max

**Comment by pangyuteng** [comment]
fyi @mrokuss  already patched in `master` branch. noice :)

https://github.com/MIC-DKFZ/nnUNet/blob/503b1d21c5e12787518c94cffe206e74f49eab9b/nnunetv2/training/lr_scheduler/polylr.py

![Image](https://github.com/user-attachments/assets/5455cf18-aa94-494a-b123-bf2f282f5e40)
